packages/ember-runtime/lib/system/core_object.js

```diff
@@ -106,87 +102,75 @@ function makeCtor() {
         FACTORY_FOR.set(this, initFactory);
         initFactory = null;
       }
-      if (initProperties) {
-        // capture locally so we can clear the closed over variable
-        let props = initProperties;
-        initProperties = null;
+
+      if (properties !== undefined && properties !== null) {
```

Comment:
Can you move the `properties !== null` into the assertion just below?  We do not want to support `Ember.Object.create(null)` at all...